### PR TITLE
Fix Service Bus messages send from HTTP failing to renew message locks

### LINKED ISSUE #157: Error returned when renewing locks on AMQP messages
Currently we use the https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-request-response#message-operations to renew locks on the active messages. 

The server response contains a field of type `0x40` which is `bin128` field type in AMQP. Vcabbage doens't currently support this type so we see an error. 

Despite the error when receiving the message I have validated with the integration test that the lock renewal works correctly. 

This is an issue to fix vcabbage and update ion.

### LINKED ISSUE #170: Blob files not passed correctly between modules
The added check in #166 caused a bug where blob SAS url's aren't correctly added to the `eventMeta` object when an event is committed.

### PR DESCRIPTION
**Pull request by lawrencegripper**
Related: #157 

So I'm very nearly finished with this fix. It's been a long journey to get to the bottom of the issue and I've done quite a bit of work with the go amqp 1 library to get things in place.

Currently I'm seeing slightly non-deterministic behavior with the renewals. The lock token is in the deliveryTag of each message but is stored in the .NET GUID byte format not the AMQP UUID format so a conversion is needed my current code only works roughly 80% of the time.  

closes #157, closes #170

### CODE REVIEW

**Comment by lawrencegripper** [comment]
This is hopefully now ready for review, end2end smoketest is passing :)

File: internal/app/dispatcher/run.go [hunk 1]
```diff
@@ -41,31 +40,56 @@ func Run(cfg *types.Configuration) {
...
 			}
 
-			err := amqpConnection.RenewLocks(ctx, messagesAMQP)
+			renewContextWithDeadline, cancel := context.WithTimeout(ctx, timeAllowanceForRenewalRequest)
+			defer cancel()
+			err := amqpConnection.RenewLocks(renewContextWithDeadline, messagesAMQP)
+			if err != nil {
+				// Todo: Additional could be put in here to cleanup operations. See: #171
+				// https://github.com/lawrencegripper/ion/issues/171
+				log.WithError(err).Panic("Failed to renew locks therefor cannot continue to operation as message maybe reassigned to another dispatcher.")
```
Comment (jjcollinge) [inline internal/app/dispatcher/run.go]: Check grammar

File: internal/app/handler/committer/committer.go [hunk 2]
```diff
@@ -150,7 +150,8 @@ func (c *Committer) commitBlob(blobsDir string) (map[string]string, error) {
 
 	logger.Info(c.context, "committed blob data")
 	logger.DebugWithFields(c.context, "blob file names", map[string]interface{}{
-		"files": files,
+		"files":    files,
+		"blobURIs": blobURIs,
```
Comment (jjcollinge) [inline internal/app/handler/committer/committer.go]: is this a potential security leak i.e. storing authenticated URLs in logs?

File: internal/app/dispatcher/run.go [hunk 3]
```diff
@@ -41,31 +40,56 @@ func Run(cfg *types.Configuration) {
 
 	var wg sync.WaitGroup
 
-	wg.Add(3)
+	wg.Add(4)
 	go func() {
 		defer wg.Done()
 		for {
+			// Locks are held for 1 mins, renew every 20 sec to keep locks
+			time.Sleep(20 * time.Second)
```
Comment (jjcollinge) [inline internal/app/dispatcher/run.go]: Should we provide cancellation contexts to our Sleeps to keep the dispatcher responsive to Kubernetes ops?

File: internal/app/dispatcher/run.go (hunk 1 above)
Comment (lawrencegripper) [reply internal/app/dispatcher/run.go]: Ta

File: internal/app/handler/committer/committer.go (hunk 2 above)
Comment (lawrencegripper) [reply internal/app/handler/committer/committer.go]: Yes, removed. Left over from debugging :(

Comment (lawrencegripper) [reply internal/app/handler/committer/committer.go]: Good spot

File: internal/pkg/servicebus/amqpconnection.go [hunk 4]
```diff
@@ -192,22 +192,45 @@ func NewAmqpConnection(ctx context.Context, config *types.Configuration) *AmqpCo
 	return &listener
 }
 
+func swapIndex(indexOne, indexTwo int, array *[16]byte) {
```
[the remark this answers is not in the document]
Comment (lawrencegripper) [reply internal/pkg/servicebus/amqpconnection.go]: Well now your just getting fancy, had gone down the explicit route but agree will fix this up.

File: internal/app/dispatcher/run.go (hunk 3 above)
Comment (lawrencegripper) [reply internal/app/dispatcher/run.go]: This is a good point - probably something worthy of a separate issue as quite a chunk of changes unrelated to this PR. It's already tracked https://github.com/lawrencegripper/ion/issues/110 as help wanted.

**Pull request merged**